### PR TITLE
fix isssue 10516 [Accessibility] CheckBox control should support ValuePattern property

### LINKED ISSUE #10516: [Accessibility] CheckBox control should support ValuePattern property
### .NET version

.Net 9.0 SDK with 9.0.100-alpha.1.23620.7 build

### Did it work in .NET Framework?

No

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

No

### Issue description

CheckBox control doesn't support ValuePattern property, compared with the CheckedListBox item, which is supported.

**CheckBox:**
image

**CheckedListBox item:**
image

### Steps to reproduce

1. Create a Winforms .Net application
2. Add a checkBox and a checkedListBox control with some items added
3. Build and run application
4. Use Accessibility Insight tool to observe the ValuePattern property for checkBox and one item of checkedListBox

### PR DESCRIPTION
**Pull request by Epica3055**
Fixes #10516 

## Proposed changes

- 
- add ValuePattern to CheckBox
- 

## Regression? 

-  No

## Risk

- minimal 

### Before

image

### After

image

## Test methodology 

- 
- manual 
- 

## Test environment(s) 
- 9.0.0-alpha.1.23623.1

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/Controls/Buttons/CheckBox.CheckBoxAccessibleObject.cs [hunk 1]
```diff
@@ -91,5 +92,47 @@ internal override void Toggle()
                 }
             }
         }
+
+        internal override void SetValue(string? newValue)
+        {
+            if (this.TryGetOwnerAs(out CheckBox? owner))
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/Controls/Buttons/CheckBox.CheckBoxAccessibleObject.cs:98]: Please invert the condition to reduce nesting.
```suggestion
            if (!this.TryGetOwnerAs(out CheckBox? owner))
            {
                return;
            }
```

**Review by Tanya-Solyanik — COMMENTED** [review]
Looks good, but please add unit tests and get this change tested by Olina's team

File: src/System.Windows.Forms/src/System/Windows/Forms/Controls/ListBoxes/CheckedListBox.cs [hunk 2]
```diff
@@ -156,7 +156,7 @@ public override int ItemHeight
     }
 
     /// <summary>
-    ///  Collection of items in this listbox.
+    ///  Collection of items in this listBox.
```
[the remark this answers is not in the document]
Comment (Tanya-Solyanik) [reply src/System.Windows.Forms/src/System/Windows/Forms/Controls/ListBoxes/CheckedListBox.cs]: This is applicable in several spots in this file

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/CheckBox.CheckBoxAccessibleObjectTests.cs [hunk 3]
```diff
@@ -287,4 +287,50 @@ public void CheckBoxAccessibleObject_GetPropertyValue_ControlType_IsExpected_For
         Assert.Equal(checkBox.AccessibilityObject.DefaultAction, ((BSTR)checkBox.AccessibilityObject.GetPropertyValue(UIA_PROPERTY_ID.UIA_LegacyIAccessibleDefaultActionPropertyId)).ToStringAndFree());
         Assert.False(checkBox.IsHandleCreated);
     }
+
+    // Unit test for https://github.com/dotnet/winforms/issues/10516.
+    [WinFormsFact]
+    public void CheckBoxAccessibleObject_Value_Pattern_Success()
+    {
+        using var checkBox = new CheckBox();
+        Assert.False(checkBox.IsHandleCreated);
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/CheckBox.CheckBoxAccessibleObjectTests.cs:296]: This assert is not needed, we hadn't performed any actions that could had created it yet.

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/CheckBox.CheckBoxAccessibleObjectTests.cs [hunk 4]
```diff
@@ -287,4 +287,50 @@ public void CheckBoxAccessibleObject_GetPropertyValue_ControlType_IsExpected_For
...
+        var checkBoxAccessibleObject = new CheckBoxAccessibleObject(checkBox);
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/CheckBox.CheckBoxAccessibleObjectTests.cs:297]: Please replace with target-types new. https://github.com/dotnet/winforms/blob/main/docs/coding-style.md#variables

**Review by Tanya-Solyanik — COMMENTED** [review]
Looks good with minor comments

**Comment by Tanya-Solyanik** [comment]
@Epica3055  - please send this fix for testing. Also it's likely that other controls have the same issues, could you please take a look at DataGridView cells and other items with check boxes. Maybe @Olina-Zhang's team can help with this investigation.

**Review by Tanya-Solyanik — APPROVED** [review]

**Review by Tanya-Solyanik — APPROVED** [review]

**Comment by Tanya-Solyanik** [comment]
@Epica3055  - could you please port this fix to release\8.0 for servicing? Thanks!

**Comment by Epica3055** [comment]
@Tanya-Solyanik @Olina-Zhang  
Since this is not a bug, so we don't have to service this pr.

**Pull request merged**